### PR TITLE
feat(agoric-cli): propose change to vault type ratio params

### PR DESCRIPTION
**Pull request by dckc**
refs: #6859

## Description

@rowgraus asked to to adjust:
 - Atom price to $13
 - Min Collat Ratio to 175%
 - Liquidation ratio to 150%
 - Vault creation fee to 0.5%

Adjusting prices is out of scope here.

But we can look at the params before changing them:

```
packages/agoric-cli$ AGORIC_NET=ollinet ./bin/agops vaults info
manager0 governance parameters:
DebtLimit [ 'IST', 1000 ]
InterestRate 1 %
LiquidationMargin 150 %
LiquidationPadding 25 %
LiquidationPenalty 1 %
LoanFee 0.5 %
```

Then we can make these proposals...

```
AGORIC_NET=ollinet ./bin/agops vaults proposeParamChange --previousOfferId $prevCharter --param LoanFee --percent 0.5 --deadline 5 >,proposeLoanFee.json
AGORIC_NET=ollinet ./bin/agoric wallet send --offer ,proposeLoanFee.json --from gov1 --keyring-backend=test

# omitting AGORIC_NET=ollinet and ./bin/ in the following

agops vaults proposeParamChange --previousOfferId $prevCharter --param LiquidationMargin --percent 150 --deadline 2 >,proposeMargin.json

agops vaults proposeParamChange --previousOfferId $prevCharter --param LiquidationPadding --percent 25 --deadline 2 >,proposePadding.json
```

... and after voting on each in the usual way:

```
$ agops vaults info
manager0 governance parameters:
DebtLimit [ 'IST', 1000 ]
InterestRate 1 %
LiquidationMargin 150 %
LiquidationPadding 25 %
LiquidationPenalty 1 %
LoanFee 0.5 %
```

### Documentation Considerations

commander `--help` doc as usual

### Testing Considerations

manually tested

### CODE REVIEW

File: packages/agoric-cli/src/commands/vaults.js [hunk 1]
```diff
@@ -3,23 +3,41 @@
...
 /** @typedef {import('@agoric/smart-wallet/src/offers').OfferSpec} OfferSpec */
 /** @typedef {import('@agoric/smart-wallet/src/offers').OfferStatus} OfferStatus */
 /** @typedef {import('@agoric/smart-wallet/src/smartWallet').BridgeAction} BridgeAction */
 
 const { vstorage, fromBoard, agoricNames } = await makeRpcUtils({ fetch });
 
+/**
+ * @param {number} manager
+ */
+const getGovernanceState = async manager => {
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:27]: not to be confused with factory governance:
```suggestion
const getManagerGovernanceState = async manager => {
```

File: packages/agoric-cli/src/commands/vaults.js [hunk 2]
```diff
@@ -59,6 +77,116 @@ export const makeVaultsCommand = async logger => {
       }
     });
 
+  vaults
+    .command('info')
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:81]: I would have expected `governance` or `params` but I see you're outputting metrics too. I can't think of a good reason someone would want only one and not the other so this lgtm. (the sometimes superfluous network request is nbd)

File: packages/agoric-cli/src/commands/vaults.js [hunk 3]
```diff
@@ -59,6 +77,116 @@ export const makeVaultsCommand = async logger => {
...
+    .description('show governance parameters of a vault type')
+    .requiredOption('--manager [N]', 'manager number', s => Number(s), 0)
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:83]: not necessary but could be nice to query factory params when the manager arg is omitted

File: packages/agoric-cli/src/commands/vaults.js [hunk 4]
```diff
@@ -59,6 +77,116 @@ export const makeVaultsCommand = async logger => {
...
+    .action(async ({ manager }) => {
+      const { entries, values } = Object;
+      /** @type {import('../lib/format.js').AssetDescriptor[]} */
+      // @ts-expect-error Brand vs. RpcRemote
+      const assets = values(agoricNames.vbankAsset).map(d => ({
+        ...d,
+        petname: d.issuerName,
+      }));
+      const fmtAmt = makeAmountFormatter(assets);
+      console.log(`manager${manager} governance parameters:`);
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:93]: process.stdout instead of logger

File: packages/agoric-cli/src/commands/vaults.js [hunk 5]
```diff
@@ -59,6 +77,116 @@ export const makeVaultsCommand = async logger => {
...
+      const { governance } = await getGovernanceState(manager);
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:94]: if we don't have a way to render a `.governance` node now's a good time to abstract that into a util in `lib/format.js`

File: packages/agoric-cli/src/commands/vaults.js [hunk 6]
```diff
@@ -59,6 +77,116 @@ export const makeVaultsCommand = async logger => {
...
+    .option(
+      '--offerId [string]',
+      'id of this offer (optional)',
+      String,
+      `proposeParamChange-${Date.now()}`,
+    )
+    .requiredOption('--manager [N]', 'manager number', s => Number(s), 0)
+    .requiredOption(
+      '--previousOfferId [string]',
+      'offer using psm charter invitation',
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:121]:
```suggestion
      'offer using EC charter invitation',
```

File: packages/agoric-cli/src/commands/vaults.js [hunk 7]
```diff
@@ -59,6 +77,116 @@ export const makeVaultsCommand = async logger => {
...
+        // TODO: amounts: 'DebtLimit'
+        'InterestRate',
+        'LiquidationMargin',
+        'LiquidationPadding',
+        'LiquidationPenalty',
+        'LoanFee',
+      ]),
+    )
+    .requiredOption(
+      '--percent <percent>',
```
Comment (turadg) [inline packages/agoric-cli/src/commands/vaults.js:134]: so this command is a `proposeRatioParamChange`?

or do you want to change the validation when it supports `debtLimit`?

File: packages/agoric-cli/src/lib/rpc.js [hunk 8]
```diff
@@ -152,14 +152,16 @@ export const boardSlottingMarshaller = (slotToVal = (s, _i) => s) => ({
     return JSON.parse(body, reviver);
   },
   serialize: whole => {
+    /** @type {Map<string, number>} */
```
Comment (turadg) [inline packages/agoric-cli/src/lib/rpc.js:155]: this is in master now accbf312ee19a7f24ba93554b3ee44b1d6c0bd8c

**Review by turadg — APPROVED** [review]
Some suggestions but nothing blocking getting in this tooling improvement

**Comment by dckc** [comment]
overtaken

**Pull request closed without merging**